[lldb][test] Build Swift Obj-C header separately

hamishknight opened the pull request:
Emitting an Obj-C header isn't supported with `-primary-file`, as the header is emitted for the whole module. Instead, split the header generation out into its own target.

hamishknight commented:
https://github.com/apple/swift/pull/32403

@swift-ci please test

adrian-prantl commented:
There is no automerger from swift/master into swift/master-next. Please also create a secondary PR with this commit cherry-picked to swift/master-next!

adrian-prantl commented on lldb/packages/Python/lldbsuite/test/make/Makefile.rules:
```diff
@@ -755,6 +752,15 @@ $(SWIFT_BRIDGING_PCH): $(SWIFT_BRIDGING_HEADER)
 
+# Swift Obj-C header.
+ifneq "$(SWIFT_OBJC_HEADER)" ""
+$(SWIFT_OBJC_HEADER):
```
I think this should be

$(SWIFT_OBJC_HEADER): $(SWIFT_SOURCES)

hamishknight replied:
Ah okay, I wasn't sure whether sources provided up-front needed to be specified as prerequisites – presumably `$(DYLIB_SWIFT_SOURCES)` should also be specified in that case?

Thanks for the review!

adrian-prantl replied:
I guess you are right. I think we should get rid of DYLIB_* and instead use a BUILD_DYLIB=YES variable. But for now, you are absolutely right!

adrian-prantl reviewed (commented):
Thank you! Looks mostly good!

hamishknight committed:
[lldb][test] Build Swift Obj-C header separately

Emitting an Obj-C header isn't supported with
`-primary-file`, as the header is emitted for
the whole module. Instead, split the header
generation out into its own target.

hamishknight commented:
https://github.com/apple/swift/pull/32403

@swift-ci please test

adrian-prantl reviewed (approved):
Please don't forget to push to swift/master-next, too.
